### PR TITLE
ird working

### PR DESCRIPTION
**Pull request by Ahesui**
Homework "Tamagotchi NFT"

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@ pub struct Tamagotchi {
     pub entertained: u64,
     pub entertained_block: u64,
     pub rested: u64,
-    pub rested_block: u64
+    pub rested_block: u64,
+    pub allowed_account: Option<ActorId>
```
Comment (shamilsan) [inline src/lib.rs:15]: You don't need to redeclare the same struct for the second time.

File: src/lib.rs [hunk 2]
```diff
@@ -136,6 +138,21 @@ extern "C" fn handle() {
             tamagotchi.sleep();
             msg::reply(TmgEvent::Slept, 0).expect("Error in reply `TmgEvent::Slept`");
         }
+        TmgAction::Transfer(new_owner) => {
+            tamagotchi.owner = new_owner;
```
Comment (shamilsan) [inline src/lib.rs:149]: Is it ok that anyone can change the owner?

File: src/lib.rs [hunk 3]
```diff
@@ -136,6 +138,21 @@ extern "C" fn handle() {
...
+            msg::reply(TmgEvent::Transfer(new_owner), 0)
+                .expect("Error in reply `TmgEvent::Transfer`");
+        }
+        TmgAction::Approve(allowed_account) => {
+            tamagotchi.allowed_account = Some(allowed_account);
```
Comment (shamilsan) [inline src/lib.rs:164]: Also, we need to add checks for the sender.

File: src/lib.rs [hunk 4]
```diff
@@ -136,6 +138,21 @@ extern "C" fn handle() {
...
+            msg::reply(TmgEvent::Approve(allowed_account), 0)
+                .expect("Error in reply `TmgEvent::Approve`");
+        }
+        TmgAction::RevokeApproval => {
+            tamagotchi.allowed_account = None;
```
Comment (shamilsan) [inline src/lib.rs]: And here.

**Comment by Ahesui** [comment]
Close first, resubmit, and fix the problems raised above

**Pull request closed without merging**